src/pykeen/losses.py

```diff
@@ -510,6 +530,199 @@ def forward(
         ))
 
 
+@parse_docdata
+class DoubleMarginLoss(PointwiseLoss):
+    r"""A limit-based scoring loss, with separate margins for positive and negative elements from [sun2018]_.
+
+    Despite its similarity to the margin-based loss, this loss is quite different to it, since it uses absolute margins
+    for positive/negative scores, rather than comparing the difference. Hence, it has a natural decision boundary
+    (somewhere between the positive and negative margin), while still resulting in sparse losses with no gradients for
+    sufficiently correct examples.
+
+    .. math ::
+        L(k, \bar{k}) = h(\bar{\lambda} + \bar{k}) + h(\lambda - k)
```

Comment:
hmm so is there a way to reformulate any pointwise loss function as a pairwise loss function? Or should we have a more general loss function that takes k, \bar{k}, and also labels?

Reply:
I guess you could consider the difference between positive and negative scores as a new score, with decision threshold 0, and then apply a pointwise loss?

Reply:
if we think about this too hard we might write half of a new paper redefining what it means to be a loss, but never finish it ;)